src/Ocelot/Multiplexer/MultiplexingMiddleware.cs

```diff
@@ -260,26 +260,24 @@ protected virtual Task MapAsync(HttpContext httpContext, Route route, List<HttpC
     protected virtual async Task<Stream> CloneRequestBodyAsync(HttpRequest request, CancellationToken aborted)
     {
         request.EnableBuffering();
-        if (request.Body.Position == 0)
+        if (request.Body.Position != 0)
         {
-            var targetBuffer = new MemoryStream();
-            if (request.ContentLength is not null)
-            {
-                await request.Body.CopyToAsync(targetBuffer, (int)request.ContentLength, aborted);
-                targetBuffer.Position = 0;
-                request.Body.Position = 0;
-            }
-            else
-            {
-                Logger.LogWarning("Aggregation does not support body copy without Content-Length header!");
-            }
+            Logger.LogWarning("Ocelot does not support body copy without stream in initial position 0");
+            return request.Body;
+        }
 
-            return targetBuffer;
+        var targetBuffer = new MemoryStream();
+        if (request.ContentLength is not null)
+        {
+            await request.Body.CopyToAsync(targetBuffer, (int)request.ContentLength, aborted);
+            targetBuffer.Position = 0;
+            request.Body.Position = 0;
         }
         else
         {
-            Logger.LogWarning("Ocelot does not support body copy without stream in initial position 0");
-            return request.Body;
+            Logger.LogWarning("Aggregation does not support body copy without Content-Length header!");
         }
+
+        return targetBuffer;

```

Comment:
[1]: https://stackoverflow.com/questions/8116541/what-exactly-is-streambuf-how-do-i-use-it ""
[2]: https://docs.unity3d.com/ScriptReference/AssetBundle.LoadFromStream.html ""
[3]: https://docs.simpleinjector.org/en/latest/lifetimes.html ""

Oops! I forgot to ask... How do we manage the `Stream` object's lifetime here?

Reply:
> @PaulARoy Regarding the management of the `Stream` object's lifetime, it appears we are relying solely on Garbage Collector to handle it.

Yes, I assume it's disposed once the request itself is disposed by aspnet

Reply:
I agree with @raman-m, it might be disposed... But since it's a clone of the original HttpContext, I'm not so sure anymore. Maybe we should try to dispose the cloned http contexts afterwards?

Reply:
### The issue has been resolved in the commit  https://github.com/ThreeMammals/Ocelot/pull/2050/commits/897c3488112d1f029639b27124831cf0556acf0f ✔️

I was wrong! The `RegisterForDispose(IDisposable)` method doesn't require implementation of `IDisposable` interface in the middleware class.